File: N-ary Tree Level Order Traversal - Leetcode 429/N-ary Tree Level Order Traversal - Leetcode 429.java
import java.util.*;


class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> output = new ArrayList<>();
        if (root == null) return output;

        Queue<Node> q = new LinkedList<>();
        q.add(root);

        while (!q.isEmpty()) {
            int n = q.size();
            List<Integer> level = new ArrayList<>();
            for (int i = 0; i < n; i++) {
                Node node = q.poll();
                level.add(node.val);
                if (node.children != null) {
                    q.addAll(node.children);
                }
            }
            output.add(level);
        }
        return output;
    }
}

File: N-ary Tree Level Order Traversal - Leetcode 429/N-ary Tree Level Order Traversal - Leetcode 429.js
var levelOrder = function(root) {
    if (!root) return [];

    let output = [];
    let q = [root];

    while (q.length > 0) {
        let n = q.length;
        let level = [];
        for (let i = 0; i < n; i++) {
            let node = q.shift();
            level.push(node.val);
            if (node.children) {
                q.push(...node.children);
            }
        }
        output.push(level);
    }
    return output;
};
